Engine/Colliders/RectangleCollider.cs

```diff
@@ -9,16 +14,118 @@ public class RectangleCollider : Collider
+        /// </param>
+        /// <returns>
+        /// True if the rectangles are in collision and false otherwise.
+        /// </returns>
+        /// <exception cref="NullReferenceException">
+        /// Throws an exception if the generated vertices list is empty.
+        /// </exception>
         public override bool checkCollision(RectangleCollider rectCollider)
         {
-            //TODO : Implement
-            throw new System.NotImplementedException();
+            // Get the rectangles' initial director vector, normal vector and vertices.
+            Vector2 direct1 = ColliderTransform.GetDirectorVector();
+            Vector2 direct2 = rectCollider.ColliderTransform.GetDirectorVector();
+
+            Vector2 normal1 = new(-direct1.Y, direct1.X);
+            Vector2 normal2 = new(-direct2.Y, direct2.X);
+
+            List<Vector2> axes = new()
+            {
+                direct1,
+                direct2,
+                normal1,
+                normal2
+            };
+
+            // Gets the rectangles' vertices list.
+            List<Vector2> vertices1 = GetVertices();
+            List<Vector2> vertices2 = rectCollider.GetVertices();
+            
+            // Check that the vertex list has been correctly created
+            if (vertices1?.Count == 0 || vertices2?.Count == 0)
+            {
+                throw new NullReferenceException("Generated vertices list is null during rectangleCollision.");
+            }
+
+            foreach (Vector2 axis in axes)
+            {
+                // Project the vertices of the first polygon onto the rectangle's normal vector
+                float minProjection1 = Vector2.Dot(normal1, vertices1[0]);
+                float maxProjection1 = minProjection1;
+
+                foreach (Vector2 vertex in vertices1)
+                {
+                    float dot = Vector2.Dot(axis, vertex);
+
+                    minProjection1 = MathF.Min(dot, minProjection1);
+                    maxProjection1 = MathF.Max(dot, maxProjection1);
+                }
+
+                // Project the vertices of the second polygon onto the rectangle's normal vector
+                float minProjection2 = Vector2.Dot(normal1, vertices2[0]);
+                float maxProjection2 = minProjection2;
+
+                foreach (Vector2 vertex in vertices2)
+                {
+                    float dot = Vector2.Dot(axis, vertex);
+
+                    minProjection2 = MathF.Min(dot, minProjection2);
+                    maxProjection2 = MathF.Max(dot, maxProjection2);
+                }
+                
+                // Check if the projected vertices overlap.
+                // If they do then according to Separating Axis Theorem, the rectangles cannot be in collision.
+                if (minProjection1 - maxProjection2 > 0 || minProjection2 - maxProjection1 > 0)
+                {
+                    return false;
+                }
+            }
```

Comment:
You can divide this code into methods to avoid repetition and also make this method more readable